core/src/banking_stage/consumer.rs

```diff
@@ -1150,10 +1162,13 @@ mod tests {
 
             //
             // TEST: When a tx in a batch can't be executed (here because of account
-            // locks), then its cost does not affect the cost tracker.
+            // locks), then its cost does not affect the cost tracker only if qos
```

Comment:
🙏 thank you for the clarification!

Reply:
It kind of feels like if we don't actually lock the account, it shouldn't go into cost tracking because we won't even attempt execution.

It shouldn't matter once we've got the scheduler *and* it's widely used, but in the mean-time should we potentially move the feature check inside the QoSUpdate, and remove any tx costs that were not locked?